internal/daemon/controller/auth/auth.go

```diff
@@ -935,3 +941,52 @@ func (r *VerifyResults) ScopesAuthorizedForList(ctx context.Context, rootScopeId
 
 	return scopeResourceMap, nil
 }
+
+// GrantsHash returns a stable hash of all the grants in the verify results.
+func (r *VerifyResults) GrantsHash(ctx context.Context) ([]byte, error) {
+	const op = "auth.GrantsHash"
+	var values []string
+	for _, grant := range r.grants {
+		values = append(values, grant.Grant, grant.RoleId, grant.ScopeId)
+	}
+	// Sort for deterministic output
+	slices.Sort(values)
+	hashVal, err := hashStrings(values...)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	return binary.LittleEndian.AppendUint64(make([]byte, 0, 4), hashVal), nil
```

Comment:
🤷🏻 we could store a uint64 representation of the hash throughout, I think this was just how it was done in the other place. It would simplify validation in some places. What do you think?

Reply:
would be more straightforward.